### PR TITLE
Add tracking

### PR DESCRIPTION
**Pull request by jamfly**
## Summary
add tracking for further analysis

## Todo

- [x] tracking at start
- [x] tracking at finish

store format
```
id: line.userId

{
    type: 'type of game', // it would be either online or O2O
    userName: 'line display name'
}
```

# Test Cases
- [x] start component
- [x] finish component

### CODE REVIEW

File: src/app/shared/final-page/final-page.component.ts [hunk 1]
```diff
@@ -51,11 +55,19 @@ export class FinalPageComponent implements OnInit, OnDestroy {
     } catch (error) {
       alert(error.message);
     }
+  }
 
+  private async trackUser(): Promise<void> {
+    const doc = this.db.collection('/finish').doc(this.profile.userId);
+    await doc.set({
+      userName: this.profile.displayName,
+      type: this.type
```
Comment (allen0304) [inline src/app/shared/final-page/final-page.component.ts]: Add 2 fields:
```
userId: this.profile.userId,
createdAt: new Date()
```

File: src/app/shared/scenario/scenario.component.ts [hunk 2]
```diff
@@ -34,11 +38,21 @@ export class ScenarioComponent implements OnInit, OnDestroy {
...
   ngOnDestroy(): void {
     this.sub.unsubscribe();
   }
 
+  private async trackUser(): Promise<void> {
+    const profile = await this.liffService.getProfile();
+    const doc = this.db.collection('/start').doc(profile.userId);
+    await doc.set({
+      userName: profile.displayName,
+      type: this.type
```
Comment (allen0304) [inline src/app/shared/scenario/scenario.component.ts]: Add 2 fields:
```
userId: this.profile.userId,
createdAt: new Date()
```

Comment (jamfly) [reply src/app/shared/scenario/scenario.component.ts]: I use `profile.userId` as  document's Id, do I have to store it again?

Comment (allen0304) [reply src/app/shared/scenario/scenario.component.ts]: yes, it's convenience to filter.

Comment (allen0304) [reply src/app/shared/scenario/scenario.component.ts]: change doc key to auto id

Comment (allen0304) [reply src/app/shared/scenario/scenario.component.ts]: By the way
use `doc.set({...}, {merge: true}) `
for tracking the latest action.

**Pull request merged**